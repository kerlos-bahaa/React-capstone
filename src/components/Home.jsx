import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { loading } from '../features/country/countrySlice';
import IconArrowRightCircle from './IconArrowRightCircle';

const Home = () => {
  const data = useSelector((state) => state.country);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loading(false));
  });

  return (
    <div className="home">
      <ul className="countries">
        {
                data.country.map((item, index) => (
                  <li key={item.name.common}>
                    <Link
                      className={((index + 2) % 2) === 0 ? 'country' : 'country  pair-red'}
                      to="/details"
                      state={{ country: item.name.common }}
                    >
                      <div className="flag-items">
                        <img className="flag" src={item.flags.png !== undefined ? item.flags.png : 'Flag not avaible'} alt={item.name.official} />
                        <IconArrowRightCircle />
                      </div>

                      <div className="linke-div">
                        <p className="crt">{item.name.common.toUpperCase()}</p>
                        <p className="crt-sub">
                          AREA :
                          {` ${item.area}`}
                        </p>
                        <p className="crt-sub">
                          POPULATION :
                          {` ${item.population}`}
                        </p>
                      </div>
                    </Link>
                  </li>
                ))
            }
      </ul>
    </div>
  );
};

export default Home;
