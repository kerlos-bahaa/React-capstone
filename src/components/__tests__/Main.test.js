import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import mockData from '../mockData';
import Home from '../Home';

jest.mock('axios');

const reducer = (state = {
  country: { country: mockData },
}) => state;

const store = configureStore({ reducer });

describe('should test MyProfile', () => {
  it('should test the snapshot of main page', async () => {
    await axios.get.mockResolvedValue({ data: mockData });
    const navb = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(navb).toMatchSnapshot();
  });
});
