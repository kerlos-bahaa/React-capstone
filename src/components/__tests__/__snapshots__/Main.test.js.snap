// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test MyProfile should test the snapshot of main page 1`] = `
<div
  className="home"
>
  <ul
    className="countries"
  >
    <li>
      <a
        className="country"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Hashemite Kingdom of Jordan"
            className="flag"
            src="https://flagcdn.com/w320/jo.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            JORDAN
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             89342
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             10203140
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country  pair-red"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Commonwealth of the Northern Mariana Islands"
            className="flag"
            src="https://flagcdn.com/w320/mp.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            NORTHERN MARIANA ISLANDS
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             464
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             57557
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Republic of Serbia"
            className="flag"
            src="https://flagcdn.com/w320/rs.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            SERBIA
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             88361
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             6908224
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country  pair-red"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Principality of Andorra"
            className="flag"
            src="https://flagcdn.com/w320/ad.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            ANDORRA
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             468
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             77265
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Turks and Caicos Islands"
            className="flag"
            src="https://flagcdn.com/w320/tc.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            TURKS AND CAICOS ISLANDS
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             948
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             38718
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country  pair-red"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Bermuda"
            className="flag"
            src="https://flagcdn.com/w320/bm.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            BERMUDA
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             54
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             63903
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        className="country"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="flag-items"
        >
          <img
            alt="Antarctica"
            className="flag"
            src="https://flagcdn.com/w320/aq.png"
          />
          <svg
            className="arrow"
            fill="currentColor"
            height="1.2em"
            viewBox="0 0 16 16"
            width="1.2em"
          >
            <path
              d="M1 8a7 7 0 1014 0A7 7 0 001 8zm15 0A8 8 0 110 8a8 8 0 0116 0zM4.5 7.5a.5.5 0 000 1h5.793l-2.147 2.146a.5.5 0 00.708.708l3-3a.5.5 0 000-.708l-3-3a.5.5 0 10-.708.708L10.293 7.5H4.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="linke-div"
        >
          <p
            className="crt"
          >
            ANTARCTICA
          </p>
          <p
            className="crt-sub"
          >
            AREA :
             14000000
          </p>
          <p
            className="crt-sub"
          >
            POPULATION :
             1000
          </p>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
